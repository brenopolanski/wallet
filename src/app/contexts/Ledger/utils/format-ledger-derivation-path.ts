import { LedgerDerivationScheme } from "app/contexts/Ledger/contracts";

export const formatLedgerDerivationPath = (scheme: LedgerDerivationScheme) =>
	`m/${scheme.purpose || 44}'/${scheme.coinType}'/${scheme.account || 0}'/${scheme.change || 0}/${
		scheme.address || 0
	}`;
