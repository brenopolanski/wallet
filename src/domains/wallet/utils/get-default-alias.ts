import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let sameCoinWalletsCount = sameCoinWallets.length || 1;

	while (profile.wallets().findByAlias(makeAlias(sameCoinWalletsCount))) {
		sameCoinWalletsCount++;
	}

	return makeAlias(sameCoinWalletsCount);
};
