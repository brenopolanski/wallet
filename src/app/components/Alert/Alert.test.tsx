import React from "react";
import { render, screen } from "utils/testing-library";

import { Alert, AlertVariant } from "./Alert";

describe("Alert", () => {
	it("should render", () => {
		const { container, asFragment } = render(<Alert />);

		expect(container).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it.each(["info", "success", "warning", "danger", "hint"])("should render as %s alert", (variant) => {
		const { container, asFragment } = render(<Alert variant={variant as AlertVariant} />);

		expect(container).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render with title", () => {
		render(<Alert title="Hello!" />);

		expect(screen.getByTestId("Alert__title")).toHaveTextContent("Hello!");
	});

	it("should render with children", () => {
		render(<Alert title="Hello!">I am a children</Alert>);

		expect(screen.getByTestId("Alert__title")).toHaveTextContent("Hello!");
		expect(screen.getByText("I am a children")).toBeInTheDocument();
	});
});
