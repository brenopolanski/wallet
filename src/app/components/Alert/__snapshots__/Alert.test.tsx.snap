// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render 1`] = `
<DocumentFragment>
  <div
    class="flex overflow-hidden flex-col space-y-4 rounded-lg p-4 bg-theme-warning-50 dark:bg-theme-secondary-800 sm:space-y-0 sm:space-x-4 sm:flex-row sm:items-center"
  >
    <div
      class="h-11 w-11 flex flex-shrink-0 items-center justify-center rounded-lg text-white bg-theme-warning-600"
    >
      <div
        class="sc-bdvvtL qGXhc"
        height="20"
        width="20"
      >
        <svg>
          circle-exclamation-mark.svg
        </svg>
      </div>
    </div>
    <div
      class="flex flex-col w-full"
    />
  </div>
</DocumentFragment>
`;

exports[`Alert should render as danger alert 1`] = `
<DocumentFragment>
  <div
    class="flex overflow-hidden flex-col space-y-4 rounded-lg p-4 bg-theme-danger-50 dark:bg-theme-secondary-800 sm:space-y-0 sm:space-x-4 sm:flex-row sm:items-center"
  >
    <div
      class="h-11 w-11 flex flex-shrink-0 items-center justify-center rounded-lg text-white bg-theme-danger-400"
    >
      <div
        class="sc-bdvvtL qGXhc"
        height="20"
        width="20"
      >
        <svg>
          circle-cross.svg
        </svg>
      </div>
    </div>
    <div
      class="flex flex-col w-full"
    />
  </div>
</DocumentFragment>
`;

exports[`Alert should render as hint alert 1`] = `
<DocumentFragment>
  <div
    class="flex overflow-hidden flex-col space-y-4 rounded-lg p-4 bg-theme-hint-50 dark:bg-theme-secondary-800 sm:space-y-0 sm:space-x-4 sm:flex-row sm:items-center"
  >
    <div
      class="h-11 w-11 flex flex-shrink-0 items-center justify-center rounded-lg text-white bg-theme-hint-500"
    >
      <div
        class="sc-bdvvtL qGXhc"
        height="20"
        width="20"
      >
        <svg>
          circle-question-mark.svg
        </svg>
      </div>
    </div>
    <div
      class="flex flex-col w-full"
    />
  </div>
</DocumentFragment>
`;

exports[`Alert should render as info alert 1`] = `
<DocumentFragment>
  <div
    class="flex overflow-hidden flex-col space-y-4 rounded-lg p-4 bg-theme-primary-50 dark:bg-theme-secondary-800 sm:space-y-0 sm:space-x-4 sm:flex-row sm:items-center"
  >
    <div
      class="h-11 w-11 flex flex-shrink-0 items-center justify-center rounded-lg text-white bg-theme-primary-600"
    >
      <div
        class="sc-bdvvtL qGXhc"
        height="20"
        width="20"
      >
        <svg>
          circle-info.svg
        </svg>
      </div>
    </div>
    <div
      class="flex flex-col w-full"
    />
  </div>
</DocumentFragment>
`;

exports[`Alert should render as success alert 1`] = `
<DocumentFragment>
  <div
    class="flex overflow-hidden flex-col space-y-4 rounded-lg p-4 bg-theme-success-50 dark:bg-theme-secondary-800 sm:space-y-0 sm:space-x-4 sm:flex-row sm:items-center"
  >
    <div
      class="h-11 w-11 flex flex-shrink-0 items-center justify-center rounded-lg text-white bg-theme-success-600"
    >
      <div
        class="sc-bdvvtL qGXhc"
        height="20"
        width="20"
      >
        <svg>
          circle-check-mark.svg
        </svg>
      </div>
    </div>
    <div
      class="flex flex-col w-full"
    />
  </div>
</DocumentFragment>
`;

exports[`Alert should render as warning alert 1`] = `
<DocumentFragment>
  <div
    class="flex overflow-hidden flex-col space-y-4 rounded-lg p-4 bg-theme-warning-50 dark:bg-theme-secondary-800 sm:space-y-0 sm:space-x-4 sm:flex-row sm:items-center"
  >
    <div
      class="h-11 w-11 flex flex-shrink-0 items-center justify-center rounded-lg text-white bg-theme-warning-600"
    >
      <div
        class="sc-bdvvtL qGXhc"
        height="20"
        width="20"
      >
        <svg>
          circle-exclamation-mark.svg
        </svg>
      </div>
    </div>
    <div
      class="flex flex-col w-full"
    />
  </div>
</DocumentFragment>
`;
